Automatically detect --disable_nhwc_to_nchw option in MO (#8450)
openvinotoolkit/openvino: docs/MO_DG/prepare_model/customize_model_optimizer/Customize_Model_Optimizer.md

are triggered by default for TensorFlow\* models only because it is the only framework with Convolution operations in
NHWC layout.

> **NOTE**: If a TensorFlow\* model is in NCHW layout, you should specify the `--disable_nhwc_to_nchw` command line
> parameter to disable these transformations.

The layout change is a complex problem and detailed explanation of it is out of this document scope. A very brief
explanation of this process is provided below:

1. Model Optimizer changes output shapes of most of operations producing 4D and 5D (four dimensional and five
dimensional) tensors as if they were in NHWC layout to NCHW layout: `nchw_shape = np.array(nhwc_shape)[0, 3, 1, 2]` for
4D and `nchw_shape = np.array(nhwc_shape)[0, 4, 1, 2, 3]` for 5D. This permutation does not happen for some operations
with specific conditions identified during a model conversion.
2. Model Optimizer inserts [Gather](../../../ops/movement/Gather_1.md) operations to the sub-graph relates to shapes
calculation to perform shape calculation in a correct layout.
3. Model Optimizer inserts [Transpose](../../../ops/movement/Transpose_1.md) operations for some operations with

> specific conditions identified during a model conversion to produce correct inference results.

Reviewer: I propose to remove only NOTE and do not remove other text because MO still change internal layout for operations and internally we execute graph in NCHW

> NHWC layout. This layout change is disabled if model doesn't have operations that OpenVINO&trade needs to execute in

Reviewer 2: ```suggestion
NHWC layout. This layout change is disabled if the model does not have operations that OpenVINO&trade needs to execute in
```

> NCHW layout, for example Convolutions in NHWC layout. It is still possible to force Model Optimizer to do layout change

Reviewer 2: ```suggestion
NCHW layout, for example, Convolutions in NHWC layout. It is still possible to force Model Optimizer to do layout change
```

> using `--disable_nhwc_to_nchw` command line parameter.

Reviewer 2: ```suggestion
using `--disable_nhwc_to_nchw` command-line parameter.
```

> > it had it before.

Reviewer: Not sure about what change it is talking about.
